android/libraries/rib-base/src/test/java/com/badoo/ribs/core/routing/configuration/ConfigurationContextTest.kt

```diff
@@ -99,32 +107,75 @@ class ConfigurationContextTest {
     }
 
     @Test
-    fun `Unresolved resolve() passes AncestryInfo & Bundles to RoutingAction with parent as default anchor `() {
+    fun `Unresolved resolve() calls RoutingAction with parent as default anchor `() {
         val parentNode = mock<Node<*>>()
-        verifyBuildNodesCalled(defaultResolver, defaultRoutingAction, parentNode, parentNode)
+        verifyBuildNodesCalledCorrectly(
+            defaultResolver,
+            defaultRoutingAction,
+            parentNode,
+            parentNode,
+            NB_EXPECTED_NODES,
+            emptyList<Bundle>()
+        )
     }
 
     @Test
-    fun `Unresolved resolve() passes AncestryInfo & Bundles to RoutingAction with expected anchor`() {
+    fun `Unresolved resolve() calls RoutingAction with expected anchor`() {
         val parentNode = mock<Node<*>>()
-        verifyBuildNodesCalled(resolverWithAnchor, routingActionWithAnchor, mockAnchor, parentNode)
+        verifyBuildNodesCalledCorrectly(
+            resolverWithAnchor,
+            routingActionWithAnchor,
+            mockAnchor,
+            parentNode,
+            NB_EXPECTED_NODES,
+            emptyList<Bundle>()
+        )
+    }
+
+    @Test
+    fun `Unresolved resolve() calls RoutingAction with proper Bundles if there's any`() {
+        val parentNode = mock<Node<*>>()
+        verifyBuildNodesCalledCorrectly(
+            defaultResolver,
+            defaultRoutingAction,
+            parentNode,
+            parentNode,
+            NB_EXPECTED_NODES,
+            MutableList(NB_EXPECTED_NODES) { mock<Bundle>() }
+        )
     }
 
-    private fun verifyBuildNodesCalled(
+
+    private fun verifyBuildNodesCalledCorrectly(
         resolver: (Parcelable) -> RoutingAction,
         routingAction: RoutingAction,
         expectedParent: Node<*>,
-        parentNode: Node<*>
+        parentNode: Node<*>,
+        nbExpectedNodes: Int,
+        bundles: List<Bundle>
     ) {
-        val bundles = emptyList<Bundle>()
-        val unresolved = Unresolved<Parcelable>(mock(), mock())
+        assertEquals("Expected empty list of bundles or exactly $nbExpectedNodes, actual: ${bundles.size}",
```

Comment:
I am pretty sure we can model this with built in asserts, but this one is fine as well